### PR TITLE
tools/cibuild.sh: Migrate risc-v toolchian from 8.3.0 to 10.2.0

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
and fix the certification expiration issue too

## Impact
See the discussion here: https://github.com/apache/incubator-nuttx/pull/6093

## Testing
Pass CI

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

File: tools/ci/cibuild.sh [hunk 1]
```diff
@@ -196,7 +196,7 @@ function riscv-gcc-toolchain {
         ;;
     esac
     cd "${prebuilt}"
-    wget --quiet https://static.dev.sifive.com/dev-tools/riscv64-unknown-elf-gcc-8.3.0-2019.08.0-${flavor}.tar.gz
+    wget --quiet https://static.dev.sifive.com/dev-tools/freedom-tools/v2020.12/riscv64-unknown-elf-toolchain-10.2.0-2020.12.8-${flavor}.tar.gz
     tar zxf riscv64-unknown-elf-gcc-8.3.0-2019.08.0-${flavor}.tar.gz
     mv riscv64-unknown-elf-gcc-8.3.0-2019.08.0-${flavor} riscv64-unknown-elf-gcc
     rm riscv64-unknown-elf-gcc-8.3.0-2019.08.0-${flavor}.tar.gz
```
Comment (pkarashchenko) [inline tools/ci/cibuild.sh:202]:
```suggestion
    tar zxf riscv64-unknown-elf-gcc-10.2.0-2020.12.8${flavor}.tar.gz
    mv riscv64-unknown-elf-gcc-10.2.0-2020.12.8${flavor} riscv64-unknown-elf-gcc
    rm riscv64-unknown-elf-gcc-10.2.0-2020.12.8${flavor}.tar.gz
```

**Review by pkarashchenko — CHANGES_REQUESTED** [review]

File: tools/ci/cibuild.sh (hunk 1 above)
Comment (xiaoxiang781216) [reply tools/ci/cibuild.sh]: Done.

**Review by pkarashchenko — APPROVED** [review]

File: tools/ci/cibuild.sh [hunk 2]
```diff
@@ -196,10 +196,10 @@ function riscv-gcc-toolchain {
         ;;
     esac
     cd "${prebuilt}"
-    wget --quiet https://static.dev.sifive.com/dev-tools/riscv64-unknown-elf-gcc-8.3.0-2019.08.0-${flavor}.tar.gz
-    tar zxf riscv64-unknown-elf-gcc-8.3.0-2019.08.0-${flavor}.tar.gz
-    mv riscv64-unknown-elf-gcc-8.3.0-2019.08.0-${flavor} riscv64-unknown-elf-gcc
-    rm riscv64-unknown-elf-gcc-8.3.0-2019.08.0-${flavor}.tar.gz
+    wget --quiet --no-check-certificate https://static.dev.sifive.com/dev-tools/freedom-tools/v2020.12/riscv64-unknown-elf-toolchain-10.2.0-2020.12.8-${flavor}.tar.gz
+    tar zxf riscv64-unknown-elf-toolchain-10.2.0-2020.12.8-${flavor}.tar.gz
+    mv riscv64-unknown-elf-toolchain-10.2.0-2020.12.8-${flavor}.tar.gz riscv64-unknown-elf-gcc
```
Comment (pkarashchenko) [inline tools/ci/cibuild.sh:201]:
```suggestion
    mv riscv64-unknown-elf-toolchain-10.2.0-2020.12.8-${flavor} riscv64-unknown-elf-gcc
```

**Review by pkarashchenko — CHANGES_REQUESTED** [review]

**Review by pkarashchenko — APPROVED** [review]

File: tools/ci/cibuild.sh (hunk 2 above)
Comment (xiaoxiang781216) [reply tools/ci/cibuild.sh]: Thanks.

**Pull request merged**